plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -219,6 +219,14 @@ public ConnectorSplitSource getSplits(
         // sort partitions
         partitions = Ordering.natural().onResultOf(HivePartition::getPartitionId).reverse().sortedCopy(partitions);
 
+        if (bucketHandle.isPresent()) {
+            if (bucketHandle.get().getReadBucketCount() > bucketHandle.get().getTableBucketCount()) {
+                throw new TrinoException(
```

Comment:
Could you please open a follow up PR dropping the `HivePartitioningHandle#maxCompatibleBucketCount`. Currently it is effectively unused as number of read buckets higher than number of table buckets is no longer supported.

Reply:
sure